### PR TITLE
add root wallet class

### PR DESCRIPTION
**Pull request by acharb**
[ticket](https://stellarorg.atlassian.net/browse/WAL-786?atlOrigin=eyJpIjoiODFiYTY4ZjUwNTk0NGQwNWEyZDc2NTgyMzQwY2ZlZTYiLCJwIjoiaiJ9)

adds the root wallet class

### CODE REVIEW

File: src/walletSdk/index.ts [hunk 1]
```diff
@@ -1,18 +1,115 @@
...
+  constructor(
+    stellarConfiguration: StellarConfiguration,
+    applicationConfiguration: ApplicationConfiguration = new ApplicationConfiguration()
+  ) {
+    this.cfg = new Config(stellarConfiguration, applicationConfiguration);
+  }
+
+  anchor(homeDomain: string, httpClientConfig: ClientConfigFn = null) {
+    const url =
+      homeDomain.indexOf("://") !== -1 ? homeDomain : `https://${homeDomain}`;
```
Comment (Ifropc) [inline src/walletSdk/index.ts:45]: With this approach there should be a configuration option in client to enable http somehow. 
In Kotlin, with one config scheme is changed to http (and I'm setting https by default in URL)
TS should has something like this too

File: src/walletSdk/Anchor/index.ts [hunk 2]
```diff
@@ -5,12 +5,15 @@ import { Auth } from "../auth";
...
 import { TomlInfo, parseToml } from "../toml";
 import { ServerRequestFailedError } from "../exception";
+import { Config } from "../";
 
 export class Anchor {
   private homeDomain = "";
+  private httpClient = null;
 
-  constructor(homeDomain) {
+  constructor(cfg: Config, homeDomain: string, httpClient) {
```
Comment (Ifropc) [inline src/walletSdk/Anchor/index.ts:14]: I'm not an expert in javascript, if class being exported, does it mean constructor is exported too? 
I saw in js SDK all "internal" classes have comments to not create them directly. We should add some comments like that later to internal constructors too.

File: src/walletSdk/index.ts [hunk 3]
```diff
@@ -1,18 +1,115 @@
...
+
+// TODO - https://stellarorg.atlassian.net/browse/WAL-789?atlOrigin=eyJpIjoiZjcwNzZhOTJmNjE1NGRhNTk1NDlkOTExMTYxMDJkZmYiLCJwIjoiaiJ9
+interface ClientConfigFn {}
+interface ClientConfig {}
+class HttpClient {}
 
-export enum NETWORKS {
-  PUBLIC = "PUBLIC",
-  TESTNET = "TESTNET",
+export class Config {
```
Comment (Ifropc) [inline src/walletSdk/index.ts:15]: This is an internal class in kt, not sure if it should be exported

File: src/walletSdk/index.ts [hunk 4]
```diff
@@ -1,18 +1,115 @@
...
+  recover(servers, httpClientConfig) {
+    return new Recovery(
+      this.cfg,
+      this.stellar(),
+      this.getClient(httpClientConfig),
+      servers
+    );
+  }
+
+  close() {
```
Comment (Ifropc) [inline src/walletSdk/index.ts:66]: Do you need to close clients explicitly in JS?

Comment (piyalbasu) [reply src/walletSdk/index.ts:66]: we shouldn;t need to explicitly close the httpclient for the JS version

File: src/walletSdk/Anchor/index.ts (hunk 2 above)
Comment (piyalbasu) [reply src/walletSdk/Anchor/index.ts:14]: Yes, the constructor will be exported, too

@Ifropc Can you please link one of the comments in js sdk you're referring to?

File: src/walletSdk/index.ts (hunk 1 above)
Comment (piyalbasu) [reply src/walletSdk/index.ts:45]: +1

File: src/walletSdk/Anchor/index.ts (hunk 2 above)
Comment (Ifropc) [reply src/walletSdk/Anchor/index.ts:14]: https://github.com/stellar/js-stellar-sdk/blob/master/src/assets_call_builder.ts#L7

Comment (piyalbasu) [reply src/walletSdk/Anchor/index.ts:14]: Gotcha - was just making sure I understood. Yeah, this makes sense. We should include similar comments in our codebase

File: src/walletSdk/index.ts (hunk 3 above)
Comment (acharb) [reply src/walletSdk/index.ts]: https://github.com/stellar/typescript-wallet-sdk/pull/5/commits/fa22147df8a55a2e01b84faeb3bf578819a0b0ef

File: src/walletSdk/index.ts (hunk 4 above)
Comment (acharb) [reply src/walletSdk/index.ts]: agreed, removed https://github.com/stellar/typescript-wallet-sdk/pull/5/commits/f6b2de78f41959406599c3bcf7dc9a9e53c02f72

File: src/walletSdk/Anchor/index.ts (hunk 2 above)
Comment (acharb) [reply src/walletSdk/Anchor/index.ts]: 👍  https://github.com/stellar/typescript-wallet-sdk/pull/5/commits/48b0bc4c95c87def43143a725f34ae15cc551b67

File: src/walletSdk/index.ts (hunk 1 above)
Comment (acharb) [reply src/walletSdk/index.ts:45]: gotchya, I can add that as part of [this ticket ](https://stellarorg.atlassian.net/browse/WAL-789?atlOrigin=eyJpIjoiYjFkOTU5ZDFkZDkxNDZhYmJhNmNiM2U3MTYxZWQyMDQiLCJwIjoiaiJ9)when adding the httpclient stuff. I made a note of it on that ticket

**Review by piyalbasu — APPROVED** [review]

**Pull request merged**